packages/flutter/lib/src/foundation/tree_diagnostics_mixin.dart

```diff
@@ -2,9 +2,1438 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
+import 'dart:ui';
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/rendering.dart';
 import 'package:meta/meta.dart';
 
-import 'print.dart';
+/// Styles for displaying a node in a [DiagnosticsNode] tree.
+///
+/// See also:
+///
+///  * [DiagnosticsNode.toStringDeep], which dumps text art trees for these
+///    styles.
+///  * [_normalTextConfiguration], which shows what text art for the standard
+///    style looks like.
+enum DiagnosticsTreeStyle {
+
+  /// Normal style for displaying trees.
+  normal,
```

Comment:
sure sparse is fine. Dart refactoring tools make iterating on this name trivial so if you come up with a better name before this code review is complete don't hesitate to suggest it.